flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/SlotStatus.java

```diff
@@ -46,13 +47,21 @@
 	/** if the slot is allocated, jobId identify which job this slot is allocated to; else, jobId is null */
 	private final JobID jobID;
 
-	public SlotStatus(SlotID slotID, ResourceProfile profiler) {
-		this(slotID, profiler, null, null);
+	/** Gateway to the TaskManager which reported the SlotStatus */
+	private final TaskExecutorGateway taskExecutorGateway;
```

Comment:
It comes with the SlotReport from the TaskExecutor. Yes, it breaks Serializable. Will change the code to contain the String address instead.